exchanges/stream/buffer/buffer_test.go

```diff
@@ -235,18 +237,16 @@ func BenchmarkNoBufferPerformance(b *testing.B) {
 		randomIndex := rand.Intn(4) //nolint:gosec // no need to import crypo/rand for testing
 		update.Asks = itemArray[randomIndex]
 		update.Bids = itemArray[randomIndex]
-		err = obl.Update(update)
-		if err != nil {
-			b.Fatal(err)
-		}
+		require.NoError(b, obl.Update(update))
 	}
 }
 
 func TestUpdates(t *testing.T) {
-	holder, _, _, err := createSnapshot()
-	if err != nil {
-		t.Error(err)
-	}
+	cp, err := getExclusivePair()
```

Comment:
Looks like you've effectively isolated tests 🎉 🌈 
Now with that said, you should be able to t.Parallel them all:
```
buffer % go test -race -count=10000
PASS
ok  	github.com/thrasher-corp/gocryptotrader/exchanges/stream/buffer	11.704s
```

Reply:
[added](https://github.com/thrasher-corp/gocryptotrader/pull/1682/commits/b9119eece4c4dc70dfbb6911fbe6afca86ca4315)